Email: Refine copy of Email Comparison pages

stephanethomas opened the pull request:
This pull request updates the copy of the `Email Comparison` page as well as the `In-Depth Comparison` page to address a number of comments made internally on Slack. I've also made a number of refinements with the aim of making this copy more consistent as well as providing users with more points of comparison, so that they can chose the most appropriate solution for their needs (some features such as real-time collaboration should make it more obvious why Google Workspace and Professional Email have different price points):


##### `Email Comparison` page

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/150798563-17b96aff-3b3d-45ff-af10-a5591c020069.png) | ![image](https://user-images.githubusercontent.com/594356/152180603-bd7e2041-fdbb-4653-abcc-3365ec1ddcaa.png)


##### `In-Depth Comparison` page (desktop)

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/150798963-e41f8cc5-80ed-460b-884d-d31452c4f129.png)<br/><br/> | ![image](https://user-images.githubusercontent.com/594356/152181018-08435bfd-06f1-4450-88ba-24b839512d5e.png)



##### `In-Depth Comparison` page (mobile)

Before | After
------ | -----
![image](https://user-images.githubusercontent.com/594356/150799441-fec9811d-c6a9-4874-a7ff-5946486907ce.png)<br/><br/><br/><br/><br/><br/><br/> | ![image](https://user-images.githubusercontent.com/594356/152181393-757f6c39-9b0d-48c3-8c42-ce1c29a4d8b7.png)


##### `Purchases` page

Note I've also updated the label of Google Workspace on the `Purchases` page, so it mentions mailboxes rather than productivity tools first:

Before | After
------ | -----
<img width="624" alt="BEFORE" src="https://user-images.githubusercontent.com/594356/150803032-d231c3fd-de56-4b3f-9691-46955567540d.png"> | <img width="623" alt="AFTER" src="https://user-images.githubusercontent.com/594356/150803043-2342e1cb-aa98-424d-ae14-e614b659b412.png">


#### Testing instructions

Note you will have to append `?flags=emails/in-depth-comparison` to the url of the page (and reload it) if you're testing those changes on a live branch as this feature has not been enabled on this environment yet:

1. Run `git checkout update/copy-in-email-comparison-pages` and start your server, or access a [live branch](https://github.com/Automattic/wp-calypso/pull/60303#issuecomment-1017725236)
2. Log into a WordPress.com account with a domain but no email
3. Open the [`Emails` page](http://calypso.localhost:3000/email)
4. Click the `Add Email` button of that domain to access the `Email Comparison` page
5. Assert that the page looks like the screenshots above
6. Click the `See how they compare` link to access the `In-Depth Comparison` page
7. Assert that the page looks like the screenshots above on desktop and mobile
8. Navigate to the [`Purchases` page](http://calypso.localhost:3000/purchases/subscriptions)
9. Assert that Google Workspace and G Suite subscriptions have the right label

stephanethomas committed:
G Suite: Update product label of Google Workspace and G Suite to highlight email before the rest

stephanethomas committed:
Emails: Refine display of list of features and app logos on Email Comparison page

stephanethomas committed:
Emails: Refine copy and list of features on Email Comparison pages

a8ci18n commented on client/lib/purchases/index.js:
```diff
@@ -673,7 +673,7 @@ function purchaseType( purchase ) {
 
 	if ( isGSuiteOrGoogleWorkspace( purchase ) ) {
-		return i18n.translate( 'Productivity Tools and Mailboxes at %(domain)s', {
+		return i18n.translate( 'Mailboxes and Productivity Tools at %(domain)s', {
```
:warning: This change will be queued for retranslation. We'll display the English text until then.

a8ci18n commented on packages/wpcom-checkout/src/checkout-labels.ts:
```diff
@@ -41,10 +41,10 @@ export function getSublabel( serverCartItem: ResponseCartProduct ): string {
 	if ( isGoogleWorkspace( serverCartItem ) || isGSuiteOrExtraLicenseProductSlug( productSlug ) ) {
 		if ( isRenewalItem ) {
-			return String( translate( 'Productivity Tools and Mailboxes Renewal' ) );
+			return String( translate( 'Mailboxes and Productivity Tools Renewal' ) );
```
:warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Productivity Tools and Mailboxes' )` *ES Score: 9*

a8ci18n commented on packages/wpcom-checkout/src/checkout-labels.ts:
```diff
@@ -41,10 +41,10 @@ export function getSublabel( serverCartItem: ResponseCartProduct ): string {
 		}
 
-		return String( translate( 'Productivity Tools and Mailboxes' ) );
+		return String( translate( 'Mailboxes and Productivity Tools' ) );
```
:warning: This change will be queued for retranslation. We'll display the English text until then.
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Productivity Tools and Mailboxes Renewal' )` *ES Score: 12*

stephanethomas committed:
Emails: Use different toggle label when list of features is shown on Email Comparison page

a8ci18n commented on client/my-sites/email/email-providers-comparison/in-depth/comparison-table.tsx:
```diff
@@ -72,7 +72,29 @@ const ComparisonTable = ( {
+
+				<tr className="email-providers-in-depth-comparison-table__separator">
+					<td className="email-providers-in-depth-comparison-table__feature">
+						{ translate( 'Collaboration' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **36** times:
`translate( 'Collaboration', { context: 'Theme Showcase term name'} )` *ES Score: 18*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/copy-in-email-comparison-pages)

a8ci18n commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -45,10 +53,24 @@ export const googleWorkspaceFeatures: EmailProviderFeatures = {
 		support: translate( '24/7 support via email' ),
-		tools: translate( 'Gmail, Calendar, Meet, Chat, Drive, Docs, Sheets, Slides and more' ),
+		tools: translate(
+			'Gmail, Calendar, Contacts, Meet, Chat, Drive, Docs, Sheets, Slides and more'
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

a8ci18n commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -45,10 +53,24 @@ export const googleWorkspaceFeatures: EmailProviderFeatures = {
 	supportUrl: ADDING_GSUITE_TO_YOUR_SITE,
+	table: {
+		access: translate( 'Online & offline' ),
+		collaboration: translate( 'Real-time' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'Real-time', { context: 'Real-time VaultPress Backups'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/copy-in-email-comparison-pages)

poligilad-auto reviewed (approved):
Looks good to me 👍 
WDYT @semihakocer?

Actually, one thing that bothers me, when adding new comparison points to emphasize why Google is a good solution, it seems like we are showing Professional email as a partial/not so good solution.

daledupreez commented on client/my-sites/email/email-provider-features/index.js:
```diff
@@ -41,7 +41,8 @@ function EmailProviderFeatures( { features, logos } ) {
 }
 
 EmailProviderFeatures.propTypes = {
-	features: PropTypes.oneOfType( [ PropTypes.node, PropTypes.string ] ),
+	features: PropTypes.arrayOf( PropTypes.oneOfType( [ PropTypes.node, PropTypes.string ] ) )
+		.isRequired,
 	logos: PropTypes.arrayOf( PropTypes.object ),
 };
```
Minor note, but it may be worth using TS-in-JSDoc comments rather than PropTypes, especially for functional components. As an example, this could be something like:
```
/**
 * @param { { features: TranslateResult[], logos?: Component[] }} features should be a list of features to list, and logos should be a list of logo objects.
 */
function EmailProviderFeatures( { features, logos } ) {
```

stephanethomas replied:
Thank you for providing an example. I think it would be better to convert this component to TypeScript but that's for another day 🙂.

daledupreez commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -29,14 +31,20 @@ export const professionalEmailFeatures: EmailProviderFeatures = {
 	},
 	supportUrl: ADDING_TITAN_TO_YOUR_SITE,
+	table: {
+		access: translate( 'Online' ),
```
I am not sure I agree with this "Online" vs "Online & offline access" as it is written -- existing emails for both services can be accessed from any email client where they have already been downloaded.

We may need to restructure this to be more specific about what kind of access we're talking about, as I _think_ we're referring to offline document edits, but I could be wrong.

semihakocer replied:
This is true, both email solutions enable offline email access. Titan webmail is still accessible without an internet connection. You can read your emails, draft new ones to send and move or star them as you please (see more https://support.titan.email/hc/en-us/articles/900001952106-Using-Titan-Webmail-like-a-Desktop-App) I think the distinction lies in the fact that you can setup offline access for all productivity tools in Google Workspace. Here's a recommendation:
For Titan:
Online & offline email access
For Google Workspace:
Online & offline Gmail, Google Drive and Calendar access

stephanethomas replied:
I've spent a bit of time experimenting with offline access with both Titan and Google Workspace, and I believe you get a decent offline experience only with the mobile apps. If you're using a browser then options are really limited as soon as you lose your Internet connection (e.g. you won't be able to browse emails that you have received unless they were loaded by the webmail while you were online). So I think I'm going to remove that comparison point given that offline support is only available in specific use cases.

daledupreez commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -45,10 +53,24 @@ export const googleWorkspaceFeatures: EmailProviderFeatures = {
 	),
 	list: {
+		access: translate( 'Online & offline access' ),
+		collaboration: translate( 'Real-time collaboration' ),
```
"Real-time collaboration" also feels a bit unclear -- I think we need to call out that it specifically refers to collaboration on Docs/Sheets/Slides, and possibly Meet/Chat. Maybe labelling the category as "Online Productivity" and the value as "Real-time collaboration" might be an approach we could use.

semihakocer replied:
Here's how Google refers to it in their pricing: `Docs, Sheets, Slides Collaborative content creation` 
We might use this directly if that makes sense. We can also tweak a bit and say `Collaborative content creation and team messaging`

stephanethomas replied:
I've updated this string to mention the apps 👍.

daledupreez commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -45,10 +53,24 @@ export const googleWorkspaceFeatures: EmailProviderFeatures = {
+		collaboration: translate( 'Real-time collaboration' ),
 		importing: translate( 'Easy to import your existing emails and contacts' ),
-		storage: translate( '30GB storage' ),
+		storage: translate( '30GB storage with shared files' ),
```
Maybe this should be `including shared files` for clarity? At a glance, "with shared files" feels ambiguous to me, because I am not sure if the shared files are included in the total or are not counted against the 30GB quota.
Or we could call the storage out as `30GB storage for emails, contacts, and calendar` for Professional Email and `30GB storage for all solutions` for Google?
It might be cleanest to just use `30GB` for both options to keep the comparison simple.

semihakocer replied:
I'd vote for keeping this one simple. We already have the learn more links with directions to support. It'd be great to clarify it there with features list. Is anyone actively updating those support guides right now @daledupreez ?

stephanethomas replied:
I still think we should make clear that Google Workspace also offers files storage. I've updated this label to the following, let me know what you think it:
> 30GB for emails and cloud storage

daledupreez replied:
> Is anyone actively updating those support guides right now?

There hasn't been any specific or concerted effort on the respective guides, but we can definitely review them and make/contribute changes.

> 30GB for emails and cloud storage

This looks good to me!

daledupreez commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -649,7 +649,7 @@ class EmailProvidersComparison extends Component {
 				badge={ poweredByTitan }
 				description={ translate(
-					'Easy-to-use email with incredibly powerful features. Manage your email and more on any device.'
+					'Integrated email solution with powerful features for your WordPress.com site.'
```
This loses the mention of mobile access, which is something we intentionally added in our last round of copy review. I think we should try to keep that in place. I also don't like the "for your WordPress.com site" wording -- that makes me feel the email solution may be a limited or too focused on WordPress.com.

stephanethomas replied:
I'll add that sentence back, and remove the mention to WordPress.com. I will also refine the description for Google Workspace so both descriptions have almost the same length.

daledupreez commented on client/my-sites/email/email-providers-stacked-comparison/email-provider-stacked-card/email-provider-stacked-features.tsx:
```diff
@@ -83,7 +83,9 @@ export const EmailProviderStackedFeaturesToggleButton = ( {
 		>
 			<span className="email-provider-stacked-features__toggle-text">
-				{ translate( 'Show all features' ) }
+				{ isRelatedContentExpanded
+					? translate( 'Hide all features' )
+					: translate( 'Show all features' ) }
```
😍

daledupreez commented on client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx:
```diff
@@ -53,8 +53,9 @@ const badge = (
-		translate( 'Inbox, calendars, and contacts' ),
+		translate( 'Send and receive from your custom domain' ),
 		translate( '30GB storage' ),
+		translate( 'Email, calendars, and contacts' ),
```
Should we call out that users can access their email from WordPress.com as a feature? We may need to wait for the embedded inbox to ship before we promote that feature.

semihakocer replied:
We can add it for now without highlighting it maybe.

stephanethomas replied:
I would vote for waiting to avoid confusing users.

semihakocer replied:
Sounds good

daledupreez reviewed (commented):
Functionally, this is working as noted, but I do have some comments on the content.

stephanethomas committed:
G Suite: Handle case where domain is unavailable in Google Workspace price component

This fixes the following error:

Uncaught TypeError: Cannot read properties of undefined (reading 'type')
    at isRegisteredDomain (registered-domains.js:13:17)
    at eval (gsuite-supported-domain.js:28:100)
    at Array.filter (<anonymous>)
    at getGSuiteSupportedDomains (gsuite-supported-domain.js:23:18)
    at hasGSuiteSupportedDomain (gsuite-supported-domain.js:45:10)
    at GoogleWorkspacePrice (google-workspace.tsx:51:127)
    at renderWithHooks (react-dom.development.js:14985:18)
    at mountIndeterminateComponent (react-dom.development.js:17811:13)
    at beginWork (react-dom.development.js:19049:16)
    at HTMLUnknownElement.callCallback (react-dom.development.js:3945:14)

stephanethomas committed:
Emails: Refine copy and alignments on Email Comparison pages

a8ci18n commented on client/my-sites/email/email-providers-stacked-comparison/provider-cards/professional-email-card.tsx:
```diff
@@ -64,7 +65,9 @@ const professionalEmailCardInformation: ProviderCardProps = {
 	showExpandButton: true,
-	description: translate( 'Integrated email solution for your WordPress.com site.' ),
+	description: translate(
+		'Integrated email solution with powerful features. Manage your email and more on any device.'
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Easy-to-use email with incredibly powerful features. Manage your email and more on any device.' )` *ES Score: 6*

a8ci18n commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -20,22 +20,31 @@ export const professionalEmailFeatures: EmailProviderFeatures = {
+		storage: translate( '30GB storage for emails' ),
 		support: translate( '24/7 support via email' ),
-		tools: translate( 'Integrated email management, Inbox, Calendar and Contacts' ),
+		tools: translate( 'Inbox, Calendar and Contacts' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Inbox, calendars, and contacts' )` *ES Score: 11*

stephanethomas replied:
I've removed `Integrated email management` as it's not a tool per see. Or maybe I don't understand what it refers to? If that means that we can manage mailboxes from Calypso then it's also the case for Google Workspace.

a8ci18n commented on client/my-sites/email/email-providers-comparison/in-depth/data.tsx:
```diff
@@ -45,10 +54,22 @@ export const googleWorkspaceFeatures: EmailProviderFeatures = {
+		collaboration: translate( 'Real-time collaboration for Docs, Sheets, and Slides' ),
 		importing: translate( 'Easy to import your existing emails and contacts' ),
-		storage: translate( '30GB storage' ),
+		storage: translate( '30GB storage for emails and cloud storage' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( '30GB or unlimited cloud storage' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/copy-in-email-comparison-pages)

stephanethomas commented on client/my-sites/email/email-provider-features/list.js:
```diff
@@ -64,7 +66,7 @@ const getTitanFeatures = ( isMonthlyProduct = true ) => {
 		translate( '30GB storage' ),
 		translate( 'Email, calendars, and contacts' ),
-		translate( 'One-click import of existing emails and contacts' ),
+		translate( '24/7 support via email' ),
```
These changes are for the old `Email Comparison` page that we only show in the domain upsell flow. I'm just making sure that we use the same list of features here.

stephanethomas commented on client/my-sites/email/email-providers-comparison/price/google-workspace.tsx:
```diff
@@ -43,6 +43,10 @@ const GoogleWorkspacePrice = ( {
+	if ( ! domain ) {
+		return <></>;
+	}
+
```
This fixes an error triggered when refreshing the `In-Depth Comparison` page.

stephanethomas commented:
@daledupreez @semihakocer : I've took your feedback into account, and refined the copy again. This is still open for debate, so do not hesitate to comment 🙂. I've also made some very minor CSS tweaks.

semihakocer commented:
The changes look good to me @stephanethomas! Thanks for all your work 🚢

daledupreez commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -612,10 +612,10 @@ class ManagePurchase extends Component {
 				  )
 				: translate(
-						'Professional email integrated with Google Meet and other productivity tools from Google.'
+						'Business email with Gmail. Include other collaboration and productivity tools from Google.'
```
Nit: I think the second sentence should be:
> Include**s** other collaboration and productivity tools from Google.

daledupreez commented on client/my-sites/email/email-providers-comparison/index.jsx:
```diff
@@ -505,7 +505,7 @@ class EmailProvidersComparison extends Component {
 				starLabel={ starLabel }
 				description={ translate(
-					'Professional email integrated with Google Meet and other productivity tools from Google.'
+					'Business email with Gmail. Include other collaboration and productivity tools from Google.'
```
As above RE "Includes" for the second sentence.

daledupreez reviewed (approved):
This LGTM! My only request is that we change two uses of "Include" to "Includes" before we request translation.

I also want to note that we have additional comments for the Professional Email features that we should look at in a follow-up PR, possibly after further discussion: p1643385695547649/1643101084.008900-slack-C01QR3S1QJH

stephanethomas committed:
Emails: Fix typo for Google Workspace description

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7124224

Thank you @stephanethomas for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7124224).
